src/basic/chase.c

```diff
@@ -370,8 +371,22 @@ int chaseat(int dir_fd, const char *path, ChaseFlags flags, char **ret_path, int
                         continue;
                 }
 
-                /* Otherwise let's see what this is. */
-                child = r = RET_NERRNO(openat(fd, first, O_CLOEXEC|O_NOFOLLOW|O_PATH));
+                /* Otherwise let's pin it by file descriptor, via O_PATH. Sounds pretty obvious to do this,
+                 * right? You just do open() with O_PATH, and there you go. But uh, it's not that
+                 * easy. open() via O_PATH does not trigger automounts, but we usually want that (except if
+                 * CHASE_NO_AUTOFS is used). But thankfully there's a way out: the newer open_tree() call,
+                 * when specified without OPEN_TREE_CLONE actually is fully equivalent to open() with O_PATH
+                 * – except for one thing: it triggers automounts.
```

Comment:
Funnily enough, `O_PATH|O_DIRECTORY` also triggers autmounts.

Reply:
but that's... not funny at all :-)

Reply:
It depends on whether automounts are possible for files. My reading is that there is an implication that is the case but I don't see anything explicitly denying it at first glance.

If that is the case / you only care about directories here then you should be able to avoid using `open_tree` for older kernels.